examples/digital-asset-manager/src/components/Assets.tsx

```diff
@@ -24,70 +24,72 @@ const Assets = ({
   const [assetUrls, setAssetUrls] = useState<ImageData[]>([]);
   const [assetCount, setAssetCount] = useState(0);
 
-  const fetchAssetsByCollectionId = () => {
-    fetch(`/.netlify/functions/collectionId`, {
-      method: "POST",
-      body: JSON.stringify({ collectionId }),
-    })
-      .then((response) => {
-        return response.json();
-      })
-      .then((result) => {
-        setAssetUrls([...result]);
-        setAssetCount(result.length);
-        setLoading(false);
+  const fetchAssetsByCollectionId = async () => {
+    try {
+      const collectionData = await fetch(`/.netlify/functions/collectionId`, {
+        method: "POST",
+        body: JSON.stringify({ collectionId }),
       });
+
+      const result = await collectionData.json();
+
+      setAssetUrls([...result]);
+      setAssetCount(result.length);
+      setLoading(false);
+    } catch (error) {
+      console.error(error);
+    }
   };
 
   useEffect(() => {
     fetchAssetsByCollectionId();
   }, []);
 
   useEffect(() => {
-    const filteredItemLength = assetUrls.filter((o) => {
-      return o["name"].toLowerCase().includes(searchTerm.toLowerCase());
+    const filteredItemLength = assetUrls.filter((assetObject) => {
+      return assetObject["name"]
+        .toLowerCase()
+        .includes(searchTerm.toLowerCase());
     });
 
     setAssetCount(filteredItemLength.length);
   }, [searchTerm]);
 
   return (
-    <>
-      <div className="assets-container">
-        {loading ? (
-          <Loader />
-        ) : (
-          <>
-            <div className="asset-item-container">
-              {assetUrls
-                .filter((o) => {
-                  return o["name"]
-                    .toLowerCase()
-                    .includes(searchTerm.toLowerCase());
-                })
-                .map((asset, index) => {
-                  return (
-                    <CatalogItem
-                      background={
-                        <img
-                          src={asset.thumbnails.thul}
-                          data-image={asset.thumbnails.webimage}
-                          className="miro-draggable draggable-item asset-image-background"
-                          draggable={false}
-                        />
-                      }
-                      name={asset.name}
-                      key={index}
-                      property_extension={asset.property_extension}
-                    />
-                  );
-                })}
-            </div>
-            <AssetCount count={assetCount} />
-          </>
-        )}
-      </div>
-    </>
+    <div className="assets-container">
+      {loading ? (
+        <Loader />
+      ) : (
+        <>
+          <div className="asset-item-container">
+            {assetUrls
+              .filter((assetObject) => {
```

Comment:
Does it make sense to do this check inside the `.map`? Just return `null` when its not needed. Will result in 1 less loop over the array elements. Only needed if this is a BIG array imho. Microwin

Reply:
Wouldn't a filter make more sense when you need to filter something out? 😄

Reply:
yeah it does, only when you have A lot of items in the array (say + 5k) this will prevent a loop of 5k items. So i dont think we need it here